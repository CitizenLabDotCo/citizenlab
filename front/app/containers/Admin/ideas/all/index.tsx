import React, { memo } from 'react';
import { adopt } from 'react-adopt';
import { isNilOrError } from 'utils/helperUtils';

// components
import PostManager, { TFilterMenu } from 'components/admin/PostManager';

// resources
import GetProjects, { GetProjectsChildProps } from 'resources/GetProjects';
import { PublicationStatus } from 'services/projects';

interface DataProps {
  projects: GetProjectsChildProps;
}

export interface Props extends DataProps {}

const IdeasTab = memo(({ projects }: Props) => {
  const defaultFilterMenu: TFilterMenu = 'projects';
  const visibleFilterMenus: TFilterMenu[] = [
    defaultFilterMenu,
    'topics',
    'statuses',
  ];

  if (!isNilOrError(projects) && projects.projectsList !== undefined) {
    return (
      <PostManager
        type="AllIdeas"
        defaultFilterMenu={defaultFilterMenu}
        visibleFilterMenus={visibleFilterMenus}
        projects={projects.projectsList}
      />
    );
  }

  return null;
});

const publicationStatuses: PublicationStatus[] = [
  'draft',
  'published',
  'archived',
];

const Data = adopt<Props>({
  projects: (
    <GetProjects
      pageSize={250}
      sort="new"
      publicationStatuses={publicationStatuses}
      filterCanModerate={true}
    />
  ),
});

export default () => (
  <Data>{(dataProps: DataProps) => <IdeasTab {...dataProps} />}</Data>
);
