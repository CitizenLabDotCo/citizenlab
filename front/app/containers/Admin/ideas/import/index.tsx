import React, { useState } from 'react';
import { UploadFile } from 'typings';
import { get } from 'lodash-es';

import { InjectedIntlProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';

// components
import FileUploader from 'components/UI/FileUploader';

// resources
import { addIdeaImportFile } from 'services/ideaFiles';

const Import = ({ intl: { formatMessage } }: InjectedIntlProps) => {
  const [files, setFiles] = useState<UploadFile[]>([]);

  var err = { file: [{ error: 'test' }, { error: 'tost' }] };

  const handleFileOnAdd = (fileToAdd: UploadFile) => {
    try {
      addIdeaImportFile(fileToAdd.base64);
      setFiles((files) => [...files, fileToAdd]);
    } catch (errors) {
      // this.setState({
      //   errors: get(errors, 'json.errors', null),
      //   processing: false,
      //   saved: false,
      //   submitState: 'error',
      // });
      err = get(errors, 'json.errors', null);
    }
  };

  return (
    <FileUploader
      id={'bulk_idea_import'}
      onFileRemove={() => {}}
      onFileAdd={handleFileOnAdd}
      apiErrors={err}
      files={files}
    />
  );
};

export default injectIntl(Import);
