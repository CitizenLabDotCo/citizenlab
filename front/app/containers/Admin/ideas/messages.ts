import { defineMessages } from 'react-intl';

export default defineMessages({
  inputManagerMetaTitle: {
    id: 'app.containers.admin.ideas.all.inputManagerMetaTitle',
    defaultMessage: 'Input manager | Participation platform of {orgName}',
  },
  inputManagerMetaDescription: {
    id: 'app.containers.admin.ideas.all.inputManagerMetaDescription',
    defaultMessage: 'Manage inputs and their statuses.',
  },
  inputManagerPageSubtitle: {
    id: 'app.containers.admin.ideas.all.inputManagerPageSubtitle',
    defaultMessage:
      'Give feedback, add topics and move inputs from one project to another',
  },
  tabManage: {
    id: 'app.containers.admin.ideas.all.tabManage',
    defaultMessage: 'Manage',
  },
});
