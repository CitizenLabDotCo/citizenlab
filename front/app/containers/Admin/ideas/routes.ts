import Loadable from 'react-loadable';
import { LoadableLoadingAdmin } from 'components/UI/Loading';
import moduleConfiguration from 'modules';

export default () => ({
  name: 'Admin Ideas',
  path: 'ideas',
  component: Loadable({
    loader: () => import('./index'),
    loading: LoadableLoadingAdmin,
    delay: 500,
  }),
  indexRoute: {
    component: Loadable({
      loader: () => import('./all'),
      loading: () => null,
      delay: 500,
    }),
  },
  childRoutes: moduleConfiguration.routes['admin.ideas'],
});
