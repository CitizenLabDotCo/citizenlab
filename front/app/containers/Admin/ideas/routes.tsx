import React, { lazy } from 'react';
import PageLoading from 'components/UI/PageLoading';

const AdminIdeasContainer = lazy(() => import('./index'));
const AdminIdeasAll = lazy(() => import('./all'));

export default () => ({
  path: 'ideas',
  element: (
    <PageLoading>
      <AdminIdeasContainer />
    </PageLoading>
  ),
  children: [
    {
      index: true,
      element: (
        <PageLoading>
          <AdminIdeasAll />
        </PageLoading>
      ),
    },
  ],
});
