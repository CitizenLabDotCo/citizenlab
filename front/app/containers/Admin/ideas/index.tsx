import React, { useState } from 'react';

// module
import { ITab } from 'typings';

// components
import HelmetIntl from 'components/HelmetIntl';
import TabbedResource from 'components/admin/TabbedResource';
import { Outlet as RouterOutlet } from 'react-router-dom';

// i18n
import messages from './messages';
import { WrappedComponentProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';

const IdeasPage = ({ intl: { formatMessage } }: WrappedComponentProps) => {
  const [tabs] = useState<ITab[]>([
    {
      label: formatMessage(messages.tabManage),
      name: 'manage',
      url: '/admin/ideas',
    },
  ]);

  const resource = {
    title: formatMessage(messages.inputManagerPageTitle),
    subtitle: formatMessage(messages.inputManagerPageSubtitle),
  };

  return (
    <>
      <TabbedResource resource={resource} tabs={tabs}>
        <HelmetIntl
          title={messages.inputManagerMetaTitle}
          description={messages.inputManagerMetaDescription}
        />
        <div id="e2e-input-manager-container">
          <RouterOutlet />
        </div>
      </TabbedResource>
    </>
  );
};

export default injectIntl(IdeasPage);
