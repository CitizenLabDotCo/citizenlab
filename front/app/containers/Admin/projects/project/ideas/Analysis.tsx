import React from 'react';
import useFeatureFlag from 'hooks/useFeatureFlag';

import {
  Icon,
  colors,
  Text,
  Button,
  Title,
  Box,
} from '@citizenlab/cl2-component-library';
import { useIntl } from 'utils/cl-intl';
import messages from './messages';

const Analysis = () => {
  const { formatMessage } = useIntl();
  const analysisEnabled = useFeatureFlag({ name: 'analysis' });
  if (!analysisEnabled) return null;
  return (
    <Box
      display="flex"
      justifyContent="space-between"
      alignItems="center"
      borderColor={colors.borderLight}
      borderRadius="3px"
      borderWidth="1px"
      borderStyle="solid"
      p="8px 16px"
      mb="36px"
    >
      <Box display="flex" gap="16px" alignItems="center">
        <Icon name="flash" width="50px" height="50px" fill={colors.orange} />
        <Box>
          <Title variant="h3">{formatMessage(messages.analysisTitle)}</Title>
          <Text>{formatMessage(messages.analysisSubtitle)}</Text>
        </Box>
      </Box>
      <Button buttonStyle="admin-dark">
        {formatMessage(messages.analysisButton)}
      </Button>
    </Box>
  );
};

export default Analysis;
