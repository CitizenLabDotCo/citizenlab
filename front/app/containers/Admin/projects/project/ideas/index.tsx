// Libraries
import React from 'react';
import styled from 'styled-components';
import messages from '../messages';
import { adopt } from 'react-adopt';

// Utils
import { FormattedMessage } from 'utils/cl-intl';

// Resources
import GetPhases, { GetPhasesChildProps } from 'resources/GetPhases';

// Components
import { SectionTitle, SectionDescription } from 'components/admin/Section';
import PostManager, { TFilterMenu } from 'components/admin/PostManager';

// resources
import { isNilOrError } from 'utils/helperUtils';
import { withRouter, WithRouterProps } from 'utils/cl-router/withRouter';
import GetProject, { GetProjectChildProps } from 'resources/GetProject';

const StyledDiv = styled.div`
  margin-bottom: 30px;
`;

interface InputProps {}

interface DataProps {
  phases: GetPhasesChildProps;
  project: GetProjectChildProps;
}

interface Props extends InputProps, DataProps {}

interface State {}

class AdminProjectIdeas extends React.PureComponent<
  Props & WithRouterProps,
  State
> {
  render() {
    const { project, phases } = this.props;
    const defaultTimelineProjectVisibleFilterMenu = 'phases';
    const defaultContinuousProjectVisibleFilterMenu = 'statuses';
    const timelineProjectVisibleFilterMenus: TFilterMenu[] = [
      defaultTimelineProjectVisibleFilterMenu,
      'statuses',
      'topics',
    ];
    const continuousProjectVisibleFilterMenus: TFilterMenu[] = [
      defaultContinuousProjectVisibleFilterMenu,
      'topics',
    ];

    return (
      <>
        <StyledDiv>
          <SectionTitle>
            <FormattedMessage {...messages.titleInputManager} />
          </SectionTitle>
          <SectionDescription>
            <FormattedMessage {...messages.subtitleInputManager} />
          </SectionDescription>
        </StyledDiv>

        {!isNilOrError(project) && (
          <PostManager
            type="ProjectIdeas"
            projectId={project.id}
            phases={phases}
            visibleFilterMenus={
              project.attributes.process_type === 'timeline'
                ? timelineProjectVisibleFilterMenus
                : continuousProjectVisibleFilterMenus
            }
            defaultFilterMenu={
              project.attributes.process_type === 'timeline'
                ? defaultTimelineProjectVisibleFilterMenu
                : defaultContinuousProjectVisibleFilterMenu
            }
          />
        )}
      </>
    );
  }
}

const Data = withRouter(
  adopt<Props>({
    phases: ({ params, render }) => (
      <GetPhases projectId={params.projectId}>{render}</GetPhases>
    ),
    project: ({ params, render }) => (
      <GetProject projectId={params.projectId}>{render}</GetProject>
    ),
  })
);

export default () => (
  <Data>{(dataProps) => <AdminProjectIdeas {...dataProps} />}</Data>
);
