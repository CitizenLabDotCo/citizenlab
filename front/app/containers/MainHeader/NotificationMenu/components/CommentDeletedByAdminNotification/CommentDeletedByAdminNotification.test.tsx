// @ts-nocheck
// libraries
import React from 'react';
import { shallow } from 'enzyme';

// component to test
import { CommentDeletedByAdminNotification } from './';

// mock utilities
import { getIdea } from 'services/__mocks__/ideas';

jest.mock('utils/cl-intl');
jest.mock('utils/analytics', () => ({ trackEventByName: () => {} }));
jest.mock('modules', () => ({ streamsToReset: [] }));

import { getNotification } from 'services/__mocks__/notifications';

describe('<CommentDeletedByAdminNotification />', () => {
  it('renders correctly with an empty idea', () => {
    const wrapper = shallow(
      <CommentDeletedByAdminNotification
        notification={getNotification(
          'commentDeletedByAdmin',
          'comment_deleted_by_admin'
        )}
        idea={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with a non empty idea', () => {
    const wrapper = shallow(
      <CommentDeletedByAdminNotification
        notification={getNotification(
          'commentDeletedByAdmin',
          'comment_deleted_by_admin'
        )}
        idea={getIdea('ideaId', 'ideaTitle')}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
