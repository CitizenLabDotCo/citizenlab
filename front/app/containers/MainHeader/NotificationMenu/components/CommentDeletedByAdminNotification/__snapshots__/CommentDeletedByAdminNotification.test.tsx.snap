// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentDeletedByAdminNotification /> renders correctly with a non empty idea 1`] = `
<NotificationWrapper
  icon="comments"
  isRead={false}
  timing="2019-04-16T14:32:17.095Z"
>
  <FormattedMessage
    defaultMessage="Your comment on '{postTitle}' has been deleted by an admin because
      {reasonCode, select,
        irrelevant {it is irrelevant}
        inappropriate {its content is inappropriate}
        other {{otherReason}}
      }
    "
    id="app.containers.NotificationMenu.commentDeletedByAdminFor"
    values={
      Object {
        "otherReason": undefined,
        "postTitle": <T />,
        "reasonCode": "irrelevant",
      }
    }
  />
</NotificationWrapper>
`;

exports[`<CommentDeletedByAdminNotification /> renders correctly with an empty idea 1`] = `
<NotificationWrapper
  icon="comments"
  isRead={false}
  timing="2019-04-16T14:32:17.095Z"
>
  <FormattedMessage
    defaultMessage="Your comment on '{postTitle}' has been deleted by an admin because
      {reasonCode, select,
        irrelevant {it is irrelevant}
        inappropriate {its content is inappropriate}
        other {{otherReason}}
      }
    "
    id="app.containers.NotificationMenu.commentDeletedByAdminFor"
    values={
      Object {
        "otherReason": undefined,
        "postTitle": <T />,
        "reasonCode": "irrelevant",
      }
    }
  />
</NotificationWrapper>
`;
